File: src/com/l2kt/gameserver/network/clientpackets/RequestRecipeShopMakeInfo.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.World;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.serverpackets.RecipeShopItemInfo;

public final class RequestRecipeShopMakeInfo extends L2GameClientPacket
{
	private int _objectId;
	private int _recipeId;
	
	@Override
	protected void readImpl()
	{
		_objectId = readD();
		_recipeId = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player player = getClient().getActiveChar();
		if (player == null)
			return;
		
		final Player manufacturer = World.getInstance().getPlayer(_objectId);
		if (manufacturer == null || manufacturer.getStoreType() != Player.StoreType.MANUFACTURE)
			return;
		
		player.sendPacket(new RecipeShopItemInfo(manufacturer, _recipeId));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestRecipeShopMakeInfo.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.model.World
import com.l2kt.gameserver.model.actor.instance.Player
import com.l2kt.gameserver.network.serverpackets.RecipeShopItemInfo

class RequestRecipeShopMakeInfo : L2GameClientPacket() {
    private var _objectId: Int = 0
    private var _recipeId: Int = 0

    override fun readImpl() {
        _objectId = readD()
        _recipeId = readD()
    }

    override fun runImpl() {
        val player = client.activeChar ?: return

        val manufacturer = World.getInstance().getPlayer(_objectId)
        if (manufacturer == null || manufacturer.storeType != Player.StoreType.MANUFACTURE)
            return

        player.sendPacket(RecipeShopItemInfo(manufacturer, _recipeId))
    }
}